CMakeLists.txt

```diff
@@ -408,10 +400,42 @@ macro(xeus_create_target target_name linkage output_name)
             target_link_libraries(${target_name} PRIVATE ${CMAKE_DL_LIBS})
         endif ()
         if (UNIX AND NOT APPLE)
-            target_link_libraries(${target_name} PRIVATE  util rt)
+            target_link_libraries(${target_name} PRIVATE util rt)
         endif ()
     endif ()
 
+    if (XEUS_EMSCRIPTEN_WASM_BUILD)
+       target_compile_definitions(${target_name} PRIVATE XEUS_EMSCRIPTEN_WASM_BUILD)
+
+       target_compile_options(${target_name}
+         PUBLIC --std=c++17
+         PUBLIC -Wno-deprecated
+         PUBLIC "SHELL: -s USE_PTHREADS=0"
+         PUBLIC "SHELL: -s DISABLE_EXCEPTION_CATCHING=0"
+       )
+       target_link_options(${target_name}
+         PUBLIC --bind
+         PUBLIC -Wno-unused-command-line-argument
+         PUBLIC "SHELL: -s MODULARIZE=1"
+         PUBLIC "SHELL: -s EXPORT_NAME=\"createXeusModule\""
+         PUBLIC "SHELL: -s EXPORT_ES6=0"
+         PUBLIC "SHELL: -s USE_ES6_IMPORT_META=0"
+         PUBLIC "SHELL: -s DEMANGLE_SUPPORT=0"
+         PUBLIC "SHELL: -s SAFE_HEAP=0"
+         PUBLIC "SHELL: -s ASSERTIONS=0"
+         PUBLIC "SHELL: -s STACK_OVERFLOW_CHECK=0"
```

Comment:
It seems that STACK_OVERFLOW_CHECK is already set to 0 if ASSERTION is 0, which also seems to be the default?